.azure_pipelines/ci-pipeline-makefile.yml

```diff
@@ -73,6 +73,7 @@ jobs:
           make clean -C $(Build.SourcesDirectory)/tests
           make clean -C $(Build.SourcesDirectory)/solutions
           sudo rm -rf $(Build.SourcesDirectory)/build/tests
+          docker system prune -f
```

Comment:
Why only intermediate images? Can we additionally use `-a` option here, that would delete all unused images.

Reply:
That probably will increase the average pipeline runtime. I am not quite sure how the docker images are cached and reused for the next image building action.